tests/api2/test_440_snmp.py

```diff
@@ -63,65 +138,235 @@ def validate_snmp_get_sysname_uses_same_ip(hostip):
-    data = results.json()
+@pytest.mark.dependency(depends=["SNMP_STARTED"])
+def test_SNMP_settings_are_preserved():
+    data = call('snmp.config')
     assert data['community'] == COMMUNITY
     assert data['traps'] == TRAPS
     assert data['contact'] == CONTACT
     assert data['location'] == LOCATION
-    assert data['v3_password'] == PASSWORD
 
 
-def test_09_get_sysname_reply_uses_same_ip():
+@pytest.mark.dependency(depends=["SNMP_STARTED"])
+def test_sysname_reply_uses_same_ip():
     validate_snmp_get_sysname_uses_same_ip(truenas_server.ip)
 
 
 @skip_ha_tests
-def test_10_ha_get_sysname_reply_uses_same_ip():
+@pytest.mark.dependency(depends=["SNMP_STARTED"])
+def test_ha_sysname_reply_uses_same_ip():
     validate_snmp_get_sysname_uses_same_ip(truenas_server.ip)
     validate_snmp_get_sysname_uses_same_ip(truenas_server.nodea_ip)
     validate_snmp_get_sysname_uses_same_ip(truenas_server.nodeb_ip)
+
+
+@pytest.mark.dependency(depends=["SNMP_STARTED"])
+def test_SNMPv3_private_user():
+    """
+    The SNMP system user should always be available
+    """
+    # Reset the systemd restart counter
+    reset_systemd_svcs("snmpd snmp-agent")
+
+    # Make sure the createUser command is not present
+    res = ssh("tail -2 /var/lib/snmp/snmpd.conf")
+    assert 'createUser' not in res
+
+    # Make sure the SNMP system user is a rwuser
+    res = ssh("cat /etc/snmp/snmpd.conf")
+    assert "rwuser snmpSystemUser" in res
+
+    # List the SNMP users and confirm the system user
+    # This also confirms the functionality of the system user
+    res = call('snmp.get_snmp_users')
+    assert "snmpSystemUser" in res
+
+
+@pytest.mark.dependency(depends=["SNMP_STARTED"])
+@pytest.mark.parametrize('payload,attrib,errmsg', [
+    ({'v3': False, 'community': ''},
+     'snmp_update.community', 'This field is required when SNMPv3 is disabled'),
+    ({'v3': True},
+     'snmp_update.v3_username', 'This field is required when SNMPv3 is enabled'),
+    ({'v3_authtype': 'AES'},
+     'snmp_update.v3_authtype', 'Invalid choice: AES'),
+    ({'v3': True, 'v3_authtype': 'MD5'},
+     'snmp_update.v3_username', 'This field is required when SNMPv3 is enabled'),
+    ({'v3_password': 'short'},
+     'snmp_update.v3_password', 'Password must contain at least 8 characters'),
+    ({'v3_privproto': 'SHA'},
+     'snmp_update.v3_privproto', 'Invalid choice: SHA'),
+    ({'v3_privproto': 'AES'},
+     'snmp_update.v3_privpassphrase', 'This field is required when SNMPv3 private protocol is specified'),
+])
+def test_v3_validators(payload, attrib, errmsg):
+    """
+    All these configuration updates should fail.
+    """
+    with pytest.raises(ValidationErrors) as ve:
+        call('snmp.update', payload)
+    if attrib:
+        assert f"{attrib}" in ve.value.errors[0].attribute
+    if errmsg:
+        assert f"{errmsg}" in ve.value.errors[0].errmsg
+
+
+@pytest.mark.dependency(name='SNMPv3_USER_ADD', depends=["SNMP_STARTED"])
```

Comment:
Configured SNMP user can also be modelled as a dependency

Reply:
With regards to dependency, what change do you think should be made?